src/content/docs/learn/window-menu.mdx

```diff
@@ -548,5 +569,372 @@ fn main() {
 </TabItem>
 </Tabs>
 
+## Menu Item Types
+
+### Basic Menu Items
+
+You can create various types of menu items:
+
+<Tabs syncKey="menu-items">
+<TabItem label="JavaScript">
+
+```javascript
+import { Menu } from '@tauri-apps/api/menu';
+
+const menu = await Menu.new({
+  items: [
+    {
+      id: 'item1',
+      text: 'Simple Item',
+    },
+    {
+      id: 'item2',
+      text: 'Item with Icon',
+      icon: 'terminal',
+    },
+    {
+      id: 'item3',
+      text: 'Item with Hotkey',
+      hotkey: 'CmdOrCtrl+Shift+A',
+    },
+  ],
+});
+```
+
+</TabItem>
+
+<TabItem label="Rust">
+
+```rust
+use tauri::menu::{Menu, MenuItem};
+
+let item1 = MenuItem::with_id(app, "item1", "Simple Item", true, None::<&str>)?;
+let item2 = MenuItem::with_id(app, "item2", "Item with Icon", true, Some("terminal"))?;
+let item3 = MenuItem::with_id(app, "item3", "Item with Hotkey", true, None::<&str>)?;
+
+let menu = Menu::with_items(app, &[&item1, &item2, &item3])?;
+```
+
+</TabItem>
+</Tabs>
+
+### Check Menu Items
+
+Check menu items can be toggled on and off:
+
+<Tabs syncKey="check-items">
+<TabItem label="JavaScript">
+
+```javascript
+import { Menu } from '@tauri-apps/api/menu';
+
+const menu = await Menu.new({
+  items: [
+    {
+      id: 'check1',
+      text: 'Check Item',
+      checked: true,
+    },
+    {
+      id: 'check2',
+      text: 'Another Check',
+      checked: false,
+    },
+  ],
+});
+```
+
+</TabItem>
+
+<TabItem label="Rust">
+
+```rust
+use tauri::menu::{Menu, CheckMenuItem};
+
+let check1 = CheckMenuItem::with_id(app, "check1", "Check Item", true, true, None::<&str>)?;
+let check2 = CheckMenuItem::with_id(app, "check2", "Another Check", true, false, None::<&str>)?;
+
+let menu = Menu::with_items(app, &[&check1, &check2])?;
+```
+
+</TabItem>
+</Tabs>
+
+### Separators
+
+Add separators to organize your menu:
+
+<Tabs syncKey="separators">
+<TabItem label="JavaScript">
+
+```javascript
+import { Menu } from '@tauri-apps/api/menu';
+
+const menu = await Menu.new({
+  items: [
+    {
+      id: 'item1',
+      text: 'Item 1',
+    },
+    {
+      type: 'Separator',
+    },
+    {
+      id: 'item2',
+      text: 'Item 2',
+    },
+  ],
+});
+```
+
+</TabItem>
+
+<TabItem label="Rust">
+
+```rust
+use tauri::menu::{Menu, MenuItem, SeparatorMenuItem};
+
+let item1 = MenuItem::with_id(app, "item1", "Item 1", true, None::<&str>)?;
+let separator = SeparatorMenuItem::new(app)?;
+let item2 = MenuItem::with_id(app, "item2", "Item 2", true, None::<&str>)?;
+
+let menu = Menu::with_items(app, &[&item1, &separator, &item2])?;
+```
+
+</TabItem>
+</Tabs>
+
+### Submenus with Icons
```

Comment:
I think we have this part covered in `#creating-a-multi-level-menu` is enough, since the syntax is basically the same as a normal menu item, we don't need to cover it in a stand alone section